app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+                    this.loadPosts(nextProps);
+                } else {
+                    this.setState({show: true});
+                }
+            });
+        }
+    }
+
+    goToThread = wrapWithPreventDoubleTap((post) => {
+        const {actions, navigator, theme} = this.props;
+        const channelId = post.channel_id;
+        const rootId = (post.root_id || post.id);
+
+        actions.loadThreadIfNecessary(rootId, channelId);
+        actions.selectPost(rootId);
+
+        const options = {
+            screen: 'Thread',
+            animated: true,
+            backButtonTitle: '',
+            navigatorStyle: {
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg
+            },
+            passProps: {
+                channelId,
+                rootId
+            }
+        };
+
+        navigator.push(options);
+    });
+
+    handleClose = () => {
+        const {actions, navigator, onClose} = this.props;
+        if (this.refs.view) {
+            this.refs.view.zoomOut().then(() => {
+                actions.selectPost('');
+                navigator.dismissModal({animationType: 'none'});
+
+                if (onClose) {
+                    onClose();
+                }
+            });
+        }
+    };
+
+    jumpToChannel = (channelId, channelDisplayName) => {
+        if (channelId) {
+            const {actions, channelTeamId, currentTeamId, navigator, onClose, theme} = this.props;
+            const currentChannelId = this.props.channelId;
+            const {
+                handleSelectChannel,
+                handleTeamChange,
+                markChannelAsRead,
+                setChannelLoading,
+                setChannelDisplayName,
+                markChannelAsViewed
+            } = actions;
+
+            if (this.refs.view) {
+                this.refs.view.zoomOut().then(() => {
+                    actions.selectPost('');
+
+                    if (onClose) {
+                        onClose();
+                    }
+
+                    navigator.resetTo({
+                        screen: 'Channel',
+                        animated: true,
+                        animationType: 'fade',
+                        navigatorStyle: {
+                            navBarHidden: true,
+                            statusBarHidden: false,
+                            statusBarHideWithNavBar: false,
+                            screenBackgroundColor: theme.centerChannelBg
+                        }
+                    });
+
+                    if (channelTeamId && currentTeamId !== channelTeamId) {
+                        handleTeamChange(channelTeamId, false);
+                    }
+
+                    setChannelLoading(channelId !== currentChannelId);
```

Comment:
We should come up with a way to have this baked into the channel/team change action automatically since we keep having to remember to call all of these wherever we change channels. Nothing to do now unless you have any good ideas for doing that

Reply:
please create a ticket to do that ;)